pytorch_lightning/accelerators/registry.py

```diff
@@ -0,0 +1,78 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import importlib
+from inspect import getmembers, isclass
+from pathlib import Path
+from typing import Any, List, Optional
+
+from pytorch_lightning.accelerators.accelerator import Accelerator
+from pytorch_lightning.utilities.exceptions import MisconfigurationException
+
+
+class _AcceleratorRegistry(dict):
+    def register(self, accelerator: Accelerator, name: Optional[str] = None, override: bool = False) -> Accelerator:
+        """Registers an accelerator mapped to a name.
+
+        Args:
+            accelerator: the accelerator to be mapped.
+            name: the name that identifies the provided accelerator.
+            override: Whether to override an existing key.
+        """
+        if name is None:
+            name = accelerator.name()
+        elif not isinstance(name, str):
+            raise TypeError(f"`name` must be a str, found {name}")
+
+        if name in self and not override:
```

Comment:
Why do we need this override? aka why do we want to allow overrides? Will it be less error prone to require register accelerator with different names?

Reply:
There could be a case where the user would wish to override the internal accelerator as well as keeping the api consistent with `register_strategies`